omod/src/main/java/org/openmrs/module/spa/SpaActivator.java

```diff
@@ -11,19 +11,57 @@
 
 import org.apache.commons.logging.Log;
 import org.apache.commons.logging.LogFactory;
+import org.openmrs.api.AdministrationService;
+import org.openmrs.api.context.Context;
 import org.openmrs.module.BaseModuleActivator;
+import org.openmrs.module.ModuleConstants;
+import org.openmrs.module.ModuleException;
+import org.openmrs.util.OpenmrsUtil;
+
+import java.io.File;
 
 public class SpaActivator extends BaseModuleActivator {
 
 	private final Log log = LogFactory.getLog(getClass());
+	private final String FRONTENT_DIRECTORY = "frontend";
+	private final String SINGLE_SPA_STATIC_FILES_DIR = "spa.frontend.directory";
 	
 	@Override
 	public void started() {
+
+		createFrontendDirectoryInAppDirectory();
 		log.info("SPA module started");
 	}
 
 	@Override
 	public void stopped() {
 		log.info("SPA module stopped");
 	}
+
+	public void createFrontendDirectoryInAppDirectory() {
+
+		AdministrationService as = Context.getAdministrationService();
+		String folderName = as.getGlobalProperty(SINGLE_SPA_STATIC_FILES_DIR,
+				FRONTENT_DIRECTORY);
+
+		// try to load the repository folder straight away.
+		File folder = new File(folderName);
+
+		// if the property wasn't a full path already, assume it was intended to be a folder in the
+		// application directory
+		if (!folder.exists()) {
+			folder = new File(OpenmrsUtil.getApplicationDataDirectory(), folderName);
+		}
+
+		// now create the modules folder if it doesn't exist
+		if (!folder.exists()) {
+			log.warn("Frontend directory " + folder.getAbsolutePath() + " doesn't exist.  Creating it now.");
+			folder.mkdirs();
+		}
+
+		if (!folder.isDirectory())
```

Comment:
Do you intentionally want to prevent openmrs from starting if there is a problem while creating this directory?

Reply:
Yeah, I think it's better to fail fast than to load fully with buggy/surprising behavior.

Reply:
@dkayiwa what do you suggest?

Reply:
@brandones says it is better to fail fast.

Reply:
Let us fix the code style https://wiki.openmrs.org/display/docs/Java+Conventions#JavaConventions-CodeStyle